### PR TITLE
Refactor "inform", etc., documentation

### PR DESCRIPTION
**Pull request by richsalz**
This adds a *big* section to openssl.pod documenting all the inform/outform/etc options.

Edit all the manpages to point to this new section.

I might have missed some, please let me know.

### CODE REVIEW

**Comment by richsalz** [comment]
I took another look; I do not think I missed anything. :)
This doesn't use the doc-template stuff (#10159) because there's not enough common text.
I *could* add a subroutine or flags or such, but that doesn't seem worth it.

**Comment by t8m** [comment]
find-doc-nits complains: doc/man1/openssl.pod:1: found 'lower-case' should use 'lowercase'

**Comment by t8m** [comment]
I am not sure you aren't deleting sometimes some useful information from the individual commands options description. For example you're sometimes removing the information that an option does not have any effect in some circumstances.

**Comment by richsalz** [comment]
I reviewed a diff against master.  I think you're right about losing some information, and I added more clarity a handful of pages.  Thanks for that!

Fixup commit pushed.

File: doc/man1/openssl-s_server.pod [hunk 1]
```diff
@@ -327,7 +329,9 @@ is in use.
 
 =item B<-dcertform> B<DER>|B<PEM>, B<-dkeyform> B<DER>|B<PEM>, B<-dpass> I<val>
 
-Additional certificate and private key format and passphrase respectively.
+The format of the certificate, private key, and passphrase for the
+additional cert; the default format is B<PEM>.
+see L<openssl(1)/Pass Phrase Options>.
```
Comment (t8m) [inline doc/man1/openssl-s_server.pod]: Perhaps separate the dpass or add both links to format options and pass phrase options.

File: doc/man1/openssl-s_server.pod [hunk 2]
```diff
@@ -344,7 +348,9 @@ B<-xcert> I<infile>, B<-xchain> options.
 
 =item B<-xcertform> B<DER>|B<PEM>, B<-xkeyform> B<DER>|B<PEM>
 
-Extra certificate and private key format respectively.
+The format for the extra certificate and private key, respectively;
+the default format is B<PEM>.
+see L<openssl(1)/Pass Phrase Options>.
```
Comment (t8m) [inline doc/man1/openssl-s_server.pod]: Wrong link here.

File: doc/man1/openssl-sess_id.pod [hunk 3]
```diff
@@ -24,6 +24,9 @@ the SSL session master key) in human readable format. Since this is a
 diagnostic tool that needs some knowledge of the SSL protocol to use
 properly, most users will not need to use it.
 
+The precise format of the data can vary across OpenSSL versions and
+is not otherwise documented.
```
Comment (t8m) [inline doc/man1/openssl-sess_id.pod]: otherwise?

File: doc/man1/openssl-sess_id.pod [hunk 4]
```diff
@@ -32,18 +35,13 @@ properly, most users will not need to use it.
...
+=item B<-inform> B<DER>|B<PEM>, B<-outform> B<DER>|B<PEM>|B<NSS>
 
-=item B<-outform> B<DER>|B<PEM>|B<NSS>
+The input and output formats; the default is PEM..
+See L<openssl(1)/Format Options> for details.
 
-This specifies the output format. The B<PEM> and B<DER> options have the same
-meaning and default as the B<-inform> option. The B<NSS> option outputs the
-session id and the master key in NSS keylog format.
+For B<NSS> output, the session ID and master key are reported in NSS "keylog"
```
Comment (t8m) [inline doc/man1/openssl-sess_id.pod]: the NSS .... format?

File: doc/man1/openssl-x509.pod [hunk 5]
```diff
@@ -376,8 +371,13 @@ retained.
...
 =item B<-keyform> B<DER>|B<PEM>
 
-Specifies the format (DER or PEM) of the private key file used in the
-B<-signkey> option.
+The key format; the default is B<PEM>.
+See L<openssl(1)/Format Options> for details.
+
+=item B<-CAform> B<DER>|B<PEM>, B<-CAkeyform> B<DER>|B<PEM>
+
+The format for the CA certificate and key; the defaul is B<PEM>.
```
Comment (t8m) [inline doc/man1/openssl-x509.pod]: default

File: doc/man1/openssl.pod [hunk 6]
```diff
@@ -516,6 +516,109 @@ parameters start with a minus sign:
...
+Several OpenSSL commands can take input or generate output in a variety
+of formats. The list of acceptable formats, and the default, is
+described in each command documentation.  The list of formats is
+described below. Both uppercase and lowercase are accepted.
+
+=over 4
+
+=item B<DER>
+
+A binary format, encoded parsed to the ASN.1 Distinguished Encoding Rules
```
Comment (t8m) [inline doc/man1/openssl.pod]: parsed?

File: doc/man1/openssl.pod [hunk 7]
```diff
@@ -516,6 +516,109 @@ parameters start with a minus sign:
...
+(DER) of the ASN.1 data language.
+
+=item B<ENGINE>
+
+Used to specify that the cryptographic material is in an OpenSSL B<engine>.
+An engine must be configured or specified using the B<-engine> option,
+and the "filename," such as might be specified with the B<-input> option,
```
Comment (t8m) [inline doc/man1/openssl.pod]: the comma inside double quotes does not look right

File: doc/man1/openssl.pod [hunk 8]
```diff
@@ -516,6 +516,109 @@ parameters start with a minus sign:
...
+
+ Text before the BEGIN line is ignored.
+ ----- BEGIN object-type -----
+ OT43gQKBgQC/2OHZoko6iRlNOAQ/tMVFNq7fL81GivoQ9F1U0Qr+DH3ZfaH8eIkX
+ xT0ToMPJUzWAn8pZv0snA0um6SIgvkCuxO84OkANCVbttzXImIsL7pFzfcwV/ERK
+ UM6j0ZuSMFOCr/lGPAoOQU0fskidGEHi1/kW+suSr28TqsyYZpwBDQ==
+ ----- END object-type -----
+ Text after the END line is also ignored
+
+The B<object-type> must match the type of object that is expected.
```
Comment (t8m) [inline doc/man1/openssl.pod]: `I<object-type>`

File: doc/man1/openssl.pod [hunk 9]
```diff
@@ -516,6 +516,109 @@ parameters start with a minus sign:
...
+ PKCS7
+ PRIVATE KEY
+ PUBLIC KEY
+ RSA PRIVATE KEY
+ SSL SESSION PARAMETERS
+ TRUSTED CERTIFICATE
+ X509 CRL
+ X9.42 DH PARAMETERS
+
+The following legacy B<object-type>'s are also supported for compatibility
```
Comment (t8m) [inline doc/man1/openssl.pod]: same as above

File: doc/man1/openssl-s_server.pod (hunk 2 above)
Comment (richsalz) [reply doc/man1/openssl-s_server.pod]: fixed by above.

File: doc/man1/openssl-sess_id.pod (hunk 4 above)
Comment (richsalz) [reply doc/man1/openssl-sess_id.pod]: I don't understand what you mean, sorry.

File: doc/man1/openssl-x509.pod (hunk 5 above)
Comment (richsalz) [reply doc/man1/openssl-x509.pod]: fixed.

File: doc/man1/openssl.pod (hunk 6 above)
Comment (richsalz) [reply doc/man1/openssl.pod]: encoded or parsed.  reworded and fixed.

File: doc/man1/openssl.pod (hunk 7 above)
Comment (richsalz) [reply doc/man1/openssl.pod]: yeah, it was clunky.  it's now this:
```
Used to specify that the cryptographic material is in an OpenSSL B<engine>.
An engine must be configured or specified using the B<-engine> option.
In addition, the B<-input> flag can be used to name a specific object in
the engine.
A password, such as the B<-passin> flag often must be specified as well.
```

File: doc/man1/openssl.pod (hunk 8 above)
Comment (richsalz) [reply doc/man1/openssl.pod]: changed.

File: doc/man1/openssl.pod (hunk 9 above)
Comment (richsalz) [reply doc/man1/openssl.pod]: "same as above" :)

**Comment by richsalz** [comment]
all but one issue addressed; fixup commit pushed.

File: doc/man1/openssl-sess_id.pod (hunk 3 above)
Comment (t8m) [reply doc/man1/openssl-sess_id.pod]: At least I learnt more English. :)

File: doc/man1/openssl-sess_id.pod (hunk 4 above)
Comment (t8m) [reply doc/man1/openssl-sess_id.pod]: I meant isn't there missing 'the'? But of course I might be wrong as non-native speaker coming from language that does not have definite and indefinite articles at all.

File: doc/man1/openssl-s_server.pod [hunk 10]
```diff
@@ -327,10 +327,15 @@ A file containing trusted certificates to use when attempting to build the
...
 
-The format of the certificate, private key, and passphrase for the
-additional cert; the default format is B<PEM>.
+The format of the certificate and private key; the default is B<PEM>
+see L<openssl(1)/Format Options>.
+
+=item B<-dpass> I<val>
+
+The passphrase for the additional private key.
+For more information about the format of B<val>
```
Comment (t8m) [inline doc/man1/openssl-s_server.pod]: `I<val>` (I am sorry :-))

File: doc/man1/openssl-sess_id.pod (hunk 4 above)
Comment (richsalz) [reply doc/man1/openssl-sess_id.pod]: No, it's correct. I  can't think of an example other than something like "report in PEM format"  And yes, it's indefinite.

Your English is better than my non-existant Czech.

File: doc/man1/openssl-s_server.pod (hunk 10 above)
Comment (richsalz) [reply doc/man1/openssl-s_server.pod]: also fixed for the -pass section a few lines above.

Comment (t8m) [reply doc/man1/openssl-s_server.pod]: You've forgot to push?

Comment (richsalz) [reply doc/man1/openssl-s_server.pod]: oops.

**Comment by t8m** [comment]
Merged to master as 777182a0c7

**Comment by richsalz** [comment]
Thanks @t8m and @levitte for your eagle-eyed reviews!

**Pull request closed without merging**